Babylon authoritarian ideologies and events

HarpagonFKowalski opened the pull request:
Journal entry rework + redone end-of-JE event + new ideologies for IG's + leader weights + localization + other things

Probably did at least one stupid thing, but most stuff should work + I did rigidly playtest base functionality (ideology weights, event modifiers and JE should work 100% of the time)

HarpagonFKowalski committed:
Three new ideologies for the Authority Rework update + localization

HarpagonFKowalski committed:
Quick fix, forgot to add '_desc' in one line.

Quick fix, forgot to add '_desc' in one line.

HarpagonFKowalski committed:
Preliminary changes to the JE Religion, autocracy, nationality (00_autocracy.txt)

Preliminary changes to the JE Religion, autocracy, nationality (00_autocracy.txt).
More to follow, including revising the completion event to work as designed and intended in the google docs for the authoritarianism rework

NOTE - current version has VFM laws commented out.

HarpagonFKowalski committed:
Working on the finishing event for the authoritarianism JE

Not fit to be merged in any capacity. Continuing work on events, ideologies and modifiers for the authority rework. WIP.

HarpagonFKowalski committed:
Fixed formatting, a new modifier, continuing work on JE event

HarpagonFKowalski committed:
JE and events done!

This commit includes:
 - Leader Ideology weights/new possible leader ideology/IG combos
- Event and modifiers
- Miscellaneous localization

HarpagonFKowalski committed:
Very slight loc changes + coded in new custom tooltips explaining the changed leader ideology weights.

Very slight loc changes + coded in new custom tooltips explaining the changed leader ideology weights.

HarpagonFKowalski committed:
Deletion of superfluous comments and WIP notes to self from the files.

HarpagonFKowalski committed:
Small changes to localization + made the ideologies request Sheeps files

Small changes to localization + made the ideologies request Sheeps files.

DOES NOT INCLUDE THE ICONS THEMSELVES

Danarca committed:
Update 00_ig_ideologies.txt

indentation

Danarca committed:
Update 00_leader_ideologies.txt

Too many/few brackets and wrongly placed code

Danarca committed:
Update 00_autocracy.txt

Indentation

Danarca committed:
Update autocracy_events.txt

Indentation

Danarca committed:
Fuck it, doing formatting from VSC

Danarca reviewed (commented):
Formatted it for you, and a few missing brackets

HiddeLekanne commented on common/ideologies/00_ig_ideologies.txt:
```diff
@@ -866,11 +774,140 @@ ideology_anti_slavery = {
+}
+
+ideology_neo_feudalistic = {
+	icon = "gfx/interface/icons/ideology_icons/ideology_leader_neo_feudalistic.dds.dds"
```
dds.dds?

HiddeLekanne commented on common/ideologies/00_leader_ideologies.txt:
```diff
@@ -2565,6 +2611,12 @@ ideology_theocrat = {
 				scope:interest_group = {
 					ig_approval <= -5
 					is_interest_group_type = ig_devout
+					OR = {
```
unnecessary OR

HiddeLekanne commented on common/ideologies/00_leader_ideologies.txt:
```diff
@@ -2565,6 +2611,12 @@ ideology_theocrat = {
+					OR = {
+						AND = {
+							is_interest_group_type = ig_intelligentsia
+							has_variable = neo_scholastic_leader_ideology_weights
```
If this is going to be a thing that is going to be reused many times, make it a scripted trigger and give it a proper name. (and a comment explaining it)

HiddeLekanne commented on common/ideologies/00_leader_ideologies.txt:
```diff
@@ -1266,31 +1300,36 @@ ideology_jingoist_leader = {
 		}
+		if = {
+			limit = {
+				scope:interest_group = {
+					is_interest_group_type = ig_trade_unions
+					has_variable = militant_populism_leader_ideology_weights
+				}
```
This needs proper names and made into scripted triggers.

HiddeLekanne commented on common/ideologies/00_leader_ideologies.txt:
```diff
@@ -147,29 +145,53 @@ ideology_traditionalist = {
+						AND = {
+							is_interest_group_type = ig_industrialists
+							has_variable = neo_feudalistic_leader_ideology_weights
+						}
```
This one too needs a scripted trigger with name and comment.

HiddeLekanne commented on common/ideologies/00_leader_ideologies.txt:
```diff
@@ -147,29 +145,53 @@ ideology_traditionalist = {
+		if = {
+			limit = {
+				scope:interest_group = {
+					AND = {
+						is_interest_group_type = ig_trade_unions
+						has_variable = militant_populism_leader_ideology_weights
+					}
```
This one too needs a scripted trigger with name and comment.

HiddeLekanne commented on common/journal_entries/00_autocracy.txt:
```diff
@@ -0,0 +1,126 @@
+﻿je_autocracy = {
+	icon = "gfx/interface/icons/event_icons/event_portrait.dds"
+	is_shown_when_inactive = {
+		OR = {
+			has_law = law_type:law_monarchy
+			has_law = law_type:law_theocracy
+		}
+		OR = {
+			has_law = law_type:law_autocracy
+			has_law = law_type:law_oligarchy
+		}
+		OR = {
+			has_law = law_type:law_state_religion
+			has_law = law_type:law_jizya
+			has_law = law:type:law_denominational_exclusion
+			#^^The above two come from VFM, can be commented out at anyones convenience
+		}
+	}
```
Is it correct that this is a and = { OR = {

Meaning that it needs to be a 
(monarchy OR theocracy) AND 
(autocracy OR oligarchy) AND 
(state_religion OR jizya OR denominational_exclusion)

HarpagonFKowalski replied:
Indeed, that is correct

HiddeLekanne commented on common/journal_entries/00_autocracy.txt:
```diff
@@ -0,0 +1,126 @@
+						has_law = law_type:law_jizya
+						has_law = law:type:law_denominational_exclusion
+						#^^The above two come from VFM, can be commented out at anyones convenience
+					}
```
Make this a scripted trigger when you use it so many times.

HiddeLekanne commented on common/journal_entries/00_autocracy.txt:
```diff
@@ -0,0 +1,126 @@
+		}
+		random_events = {
+			10 = 0
+			1 = autocracy.2			#Secret Societies event. Punish or let be an IG against authoritarianism - Intelligentsia only?
+			1 = autocracy.3			#Blessed are the Meek. Extra loyalists/less radicals from SoL changes
```
8.3% are you sure you don't mean like 10% or something? (would be 8 1 1)

HarpagonFKowalski replied:
This part comes from the vanilla version, random_event probability included, I merely added in the comments myself as a way to keep track what is what. Any potential changes there would come in the future, but not now

HiddeLekanne commented on common/journal_entries/00_autocracy.txt:
```diff
@@ -0,0 +1,126 @@
+		value = root.var:autocracy_var
+	}
+	goal_add_value = {
+		add = 360
```
30 years?

HarpagonFKowalski replied:
That 30-year-timeframe is, too, part of the vanilla JE

Me and Fallonar wanted to talk over alternative methods to progress this JE, as to make it less of a passive experience, but that is once again something to be done in the foreseeable future

HiddeLekanne commented on common/modifiers/babylon_modifiers.txt:
```diff
@@ -236,4 +236,22 @@ player_construction_bonus = {
+}
+
+#Below you will find things used by the Authoritarianism rework.
+rise_of_neo_scholasticism_mod = {
```
The modifiers sound like they should boost attraction, but they boost strength is this correct?

HarpagonFKowalski replied:
That is correct. Modifiers were implemented based on the Google Doc about Auth. Rework

HiddeLekanne commented on events/autocracy_events.txt:
```diff
@@ -0,0 +1,321 @@
+		# triggered by je_autocracy completion
+	}
+	immediate = {
+		ruler = {
+			save_scope_as = monarch_scope
+		}
+		random_primary_culture = {
+			save_scope_as = primary_culture_scope
+		}
+		ig:ig_devout = {
+			save_scope_as = church_scope
+		}
+		ig:ig_landowners = {
+			save_scope_as = landowners_ig
+		}
+		ig:ig_industrialists = {
+			save_scope_as = industrialist_ig
+		}
+		ig:ig_armed_forces = {
+			save_scope_as = military_ig
+		}
+		ig:ig_intelligentsia = {
+			save_scope_as = intelligentsia_ig
+		}
+		ig:ig_trade_unions = {
+			save_scope_as = trade_unions_ig
+		}
```
Unused (and useless because they don't contain any extra information) scopes?

HarpagonFKowalski replied:
Some of them are genuinely useless now, as they were used by the vanilla version of Events, but aren't used by our re-imagined versions. They will be deleted

HiddeLekanne commented on events/autocracy_events.txt:
```diff
@@ -0,0 +1,321 @@
+			add_ideology = ideology_militant_populism
+			set_variable = {
+				name = militant_populism_leader_ideology_weights
+			}
```
why not check for ideology later? Now you add extra variable which tracks the ideology, which is already easily accessible? 

Unless you plan to have more transitions towards the same ideology and the kind of transition being relevant later? I don't see that currently.

HiddeLekanne commented on common/ideologies/00_ig_ideologies.txt:
```diff
@@ -866,11 +774,140 @@ ideology_anti_slavery = {
 	}
 }
+
+ideology_neo_scholastic = {
```
ideologies we add ourselves should go into babylon_ig_ideologies.txt . If it doesn't exist we create one.

HiddeLekanne commented on events/autocracy_events.txt:
```diff
@@ -0,0 +1,321 @@
+			add_ideology = ideology_neo_feudalistic
+			set_variable = {
+				name = neo_feudalistic_leader_ideology_weights
+			}
```
why not check for ideology later? Now you add extra variable which tracks the ideology, which is already easily accessible?

Unless you plan to have more transitions towards the same ideology and the kind of transition being relevant later? I don't see that currently.

HiddeLekanne commented on events/autocracy_events.txt:
```diff
@@ -0,0 +1,321 @@
+			months = long_modifier_time
+		}
+	}
+	option = {
```
no aristocrats loyalists?

HarpagonFKowalski replied:
Good call, will be added in

HiddeLekanne commented on events/autocracy_events.txt:
```diff
@@ -0,0 +1,321 @@
+				}
+			}
+			save_scope_as = secret_societies_state
+			random_scope_pop = {
+				limit = {
+					is_pop_type = academics
+				}
+				save_scope_as = secret_society_pop
+			}
```
Unused scope? also problematic scope because there probably will be multiple academic pops with slightly different religions and/or cultures.

HarpagonFKowalski replied:
Part of the vanilla JE, not our work. I am fairly sure the scope is used by one of the two random events, the hidden societies one

HiddeLekanne commented on localization/english/content_1_l_english.yml:
why are we adding this vanilla file?

HarpagonFKowalski committed:
working on changes suggested by Kikker

HiddeLekanne commented on localization/english/old content ignore.yml:
delete the file if not used anymore

HarpagonFKowalski committed:
Further work on Kikkers suggestions

Deleted unused files, fixed the localization issues (removed: an entire vanilla file. Added: one small file to replacements + a few lines to pre-existing babylon_ files)

HarpagonFKowalski committed:
WIP on scripted triggers

HarpagonFKowalski committed:
Further WIP on triggers

Reworked how weights are generated, added a new trigger (is_authoritarian_IG)

HarpagonFKowalski committed:
(Hopefully) fixed formatting + missing bracket

HarpagonFKowalski committed:
remove every_interest_group

HarpagonFKowalski committed:
bracket fix

HarpagonFKowalski committed:
tab fixes : )

HarpagonFKowalski committed:
removed useless OR

HiddeLekanne committed:
Merge branch 'Babylon-Core' into Babylon-authoritarian-ideologies-and-events
